scripts/build/gen_isr_tables.py

```diff
@@ -14,142 +14,227 @@
+        #              that directly go to the processor).
+        # 0x0000FF00 - represents the 2nd level (i.e. the interrupts funnel
+        #              into 1 line which then goes into the 1st level)
+        # 0x00FF0000 - represents the 3rd level (i.e. the interrupts funnel
+        #              into 1 line which then goes into the 2nd level)
+        self.__int_lvl_masks = [0x000000FF, 0x0000FF00, 0x00FF0000]
+
+        self.__irq2_baseoffset = None
+        self.__irq3_baseoffset = None
+        self.__irq2_offsets = None
+        self.__irq3_offsets = None
+
+        if self.check_multi_level_interrupts():
+            self.__max_irq_per = self.get_sym("CONFIG_MAX_IRQ_PER_AGGREGATOR")
+
+            self.__int_bits[0] = self.get_sym("CONFIG_1ST_LEVEL_INTERRUPT_BITS")
+            self.__int_bits[1] = self.get_sym("CONFIG_2ND_LEVEL_INTERRUPT_BITS")
+            self.__int_bits[2] = self.get_sym("CONFIG_3RD_LEVEL_INTERRUPT_BITS")
+
+            if sum(self.int_bits) > 32:
+                raise ValueError("Too many interrupt bits")
+
+            self.__int_lvl_masks[0] = self.__bm(self.int_bits[0])
+            self.__int_lvl_masks[1] = self.__bm(self.int_bits[1]) << self.int_bits[0]
+            self.__int_lvl_masks[2] = self.__bm(self.int_bits[2]) << (self.int_bits[0] + self.int_bits[1])
+
+            self.__log.debug("Level    Bits        Bitmask")
+            self.__log.debug("----------------------------")
+            for i in range(3):
+                bitmask_str = "0x" + format(self.__int_lvl_masks[i], '08X')
+                self.__log.debug(f"{i + 1:>5} {self.__int_bits[i]:>7} {bitmask_str:>14}")
+
+            if self.check_sym("CONFIG_2ND_LEVEL_INTERRUPTS"):
+                num_aggregators = self.get_sym("CONFIG_NUM_2ND_LEVEL_AGGREGATORS")
+                self.__irq2_baseoffset = self.get_sym("CONFIG_2ND_LVL_ISR_TBL_OFFSET")
+                self.__irq2_offsets = [self.get_sym('CONFIG_2ND_LVL_INTR_{}_OFFSET'.
+                                                  format(str(i).zfill(2))) for i in
+                                     range(num_aggregators)]
+
+                self.__log.debug('2nd level offsets: {}'.format(self.__irq2_offsets))
+
+                if self.check_sym("CONFIG_3RD_LEVEL_INTERRUPTS"):
+                    num_aggregators = self.get_sym("CONFIG_NUM_3RD_LEVEL_AGGREGATORS")
+                    self.__irq3_baseoffset = self.get_sym("CONFIG_3RD_LVL_ISR_TBL_OFFSET")
+                    self.__irq3_offsets = [self.get_sym('CONFIG_3RD_LVL_INTR_{}_OFFSET'.
+                                                      format(str(i).zfill(2))) for i in
+                                         range(num_aggregators)]
+
+                    self.__log.debug('3rd level offsets: {}'.format(self.__irq3_offsets))
+
+    @property
+    def args(self):
+        return self.__args
+
+    @property
+    def swt_spurious_handler(self):
+        return self.__swt_spurious_handler
```

Comment:
Yes. It might be a little overkill to use properties here. A lot of this code was created just trying to do it "good", and then used. I was trying just to close access to some internal variables, allowing only to read them. We can get rid of this - but what would be really a point other than removing 3 lines of code for every variable?